astropy/table/tests/test_mixin.py

```diff
@@ -110,13 +110,14 @@ def test_io_write_fail(mixin_cols):
     this just confirms no exceptions.
     """
     t = QTable(mixin_cols)
-    for fmt in ('fits', 'votable', 'hdf5'):
-        if fmt == 'hdf5' and not HAS_H5PY:
-            continue
-        out = StringIO()
-        with pytest.raises(ValueError) as err:
-            t.write(out, format=fmt)
-        assert 'cannot write table with mixin column(s)' in str(err.value)
+    if all(not isinstance(col, Quantity) for col in t.columns.values()):
```

Comment:
Rewrite this as:
```
# Only do this test if there are unsupported columns types (i.e. anything besides
# BaseColumn or Quantity subclasses for the FITS, votable, and HDF5 writers).
unsupported_cols = t.columns.not_isinstance(BaseColumn, Quantity)
if not unsupported_cols:
    pytest.skip("no unsupported column types")
```
In this way you minimize the diff and also generate an explicit pytest skip.